test/integration/master_benchmark_test.go

```diff
@@ -0,0 +1,203 @@
+// +build benchmark,!no-etcd,!integration
+
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package integration
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/fields"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	etcdgeneric "github.com/GoogleCloudPlatform/kubernetes/pkg/registry/generic/etcd"
+	"github.com/GoogleCloudPlatform/kubernetes/test/integration/framework"
+	"github.com/golang/glog"
+)
+
+// Command line flag globals, parsed in init and used by the benchmarks:
+//	* pods && !workers: Start -pods, scale number of workers with b.N
+//  * !pods && workers: Start -workers, scale pods with b.N
+//  * pods && workers: Ignore b.N, benchmark behaves like a test constrained by -benchtime.
+//  * !pods && !workers: scale pods and workers with b.N.
+var (
+	Workers        int
+	Pods           int
+	RegistryPrefix string
+)
+
+const Glog_fatalf = 3
+
+func init() {
+	q := flag.Int("q", 3, "Quietness, don't glog severities <= given level.")
+	pods := flag.Int("pods", -1, "Number of pods. If unspecified, uses b.N.")
```

Comment:
Since you share the flag namespace with all the other test files, I'd recommend more verbose flag names, e.g., start with bench- or some such. Also make the descriptions more clear that they're about these benchmarks.